MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -281,6 +281,47 @@ class RouteControllerTests: XCTestCase {
         XCTAssertTrue(eventsManagerSpy.hasFlushedEvent(with: expectedEventName))
     }
     
+    func testNoReroutesAfterArriving() {
+        let routeController = dependencies.routeController
+        let firstLocation = dependencies.firstLocation
+        let lastLocation = dependencies.lastLocation
+        
+        // MARK: When navigation begins with a location update
+        routeController.locationManager(routeController.locationManager, didUpdateLocations: [firstLocation])
+        
+        // MARK: It queues and flushes a Depart event
+        XCTAssertTrue(eventsManagerSpy.hasFlushedEvent(with: MMEEventTypeNavigationDepart))
+        // TODO: should there be a delegate message here as well?
+        
+        // MARK: Update to last step
+        routeController.advanceStepIndex(to: routeController.routeProgress.route.legs.first!.steps.count - 1)
```

Comment:
As we talked about before I'm more interested in doing this kind of "journey test" of delegation and telemetry events using location updates only. I have a commit which adjusts that other test but it occurs to me that I haven't pushed it to a PR yet.

https://github.com/mapbox/mapbox-navigation-ios/commit/6a9abcf967d7637540eb3b184c93c8d04fb2c185

We can take this as-is for now and update it later, since it took a bit of a test refactor to make the intent clear.